tokens/contracts/access/MintControl.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract MintControl is Initializable, OwnableUpgradeable {
+    mapping (address => bool) private _minters;
+    bool public minterControlEnabled;
+
+    event MinterControlEnable(address indexed sender);
+    event MinterControlDisable(address indexed sender);
+    event MinterGranted(address indexed account, address indexed sender);
+    event MinterRevoked(address indexed account, address indexed sender);
+
+    function __MintControl_init() internal initializer {
+        __Ownable_init_unchained();
+        __MintControl_init_unchained();
+    }
+
+    function __MintControl_init_unchained() internal initializer {
```

Comment:
I'm not sure how these init functions work, but I guess we should ensure that if there is an update, we can still keep the same state `_minters` and `minterControlEnabled`. I remember some "hacks" with a gap variable to make sure we can do updates.

Can we have some tests for that even just manual tests